zcash_client_sqlite/src/lib.rs

```diff
@@ -253,23 +244,32 @@ impl<P: consensus::Parameters> WalletRead for WalletDb<P> {
         #[allow(deprecated)]
         wallet::transact::select_spendable_sapling_notes(self, account, target_value, anchor_height)
     }
-}
 
-#[cfg(feature = "transparent-inputs")]
-impl<P: consensus::Parameters> WalletReadTransparent for WalletDb<P> {
     fn get_transparent_receivers(
         &self,
-        account: AccountId,
+        _account: AccountId,
     ) -> Result<HashSet<TransparentAddress>, Self::Error> {
-        wallet::get_transparent_receivers(&self.params, &self.conn, account)
+        #[cfg(feature = "transparent-inputs")]
+        return wallet::get_transparent_receivers(&self.params, &self.conn, _account);
+
+        #[cfg(not(feature = "transparent-inputs"))]
+        return Err(SqliteClientError::BackendError(
+            Error::TransparentInputsNotSupported,
+        ));
```

Comment:
You can avoid `return`s by wrapping the first half in a scope:
```suggestion
        #[cfg(feature = "transparent-inputs")]
        {
            wallet::get_transparent_receivers(&self.params, &self.conn, _account)
        }

        #[cfg(not(feature = "transparent-inputs"))]
        Err(SqliteClientError::BackendError(
            Error::TransparentInputsNotSupported,
        ))
```

Reply:
I actually think that the `return` is helpful here, because it make the "the following code may not be executed" situation a bit more obvious. In general I don't like early `return`, but this is an exception.